src/main/java/com/powsybl/openloadflow/ac/outerloop/AreaInterchangeControlOuterloop.java

```diff
@@ -0,0 +1,219 @@
+            return new OuterLoopResult(this, OuterLoopStatus.UNSTABLE);
+        }
+    }
+
+    private OuterLoopResult distributionFailureResult(AcOuterLoopContext context, String areaMismatchesString, boolean movedBuses, DistributedSlackContextData contextData, double totalDistributedActivePower) {
+        String statusText = MessageFormat.format("Failed to distribute interchange active power mismatch. Remaining mismatches: [{0}]",
+                areaMismatchesString);
+        OpenLoadFlowParameters.SlackDistributionFailureBehavior slackDistributionFailureBehavior = context.getLoadFlowContext().getParameters().getSlackDistributionFailureBehavior();
+        if (OpenLoadFlowParameters.SlackDistributionFailureBehavior.DISTRIBUTE_ON_REFERENCE_GENERATOR == slackDistributionFailureBehavior) {
+            LOGGER.error("Distribute on reference generator is not supported in AreaInterchangeControlOuterloop, falling back to FAIL mode");
+            slackDistributionFailureBehavior = OpenLoadFlowParameters.SlackDistributionFailureBehavior.FAIL;
+        }
+
+        switch (slackDistributionFailureBehavior) {
+            case THROW ->
+                throw new PowsyblException(statusText);
+
+            case LEAVE_ON_SLACK_BUS -> {
+                LOGGER.warn("{}", statusText);
+                return new OuterLoopResult(this, movedBuses ? OuterLoopStatus.UNSTABLE : OuterLoopStatus.STABLE);
+            }
+            case FAIL -> {
+                LOGGER.error("{}", statusText);
+                // Mismatches reported in LoadFlowResult on slack bus(es) are the mismatches of the last NR run.
+                // Since we will not be re-running an NR, revert distributedActivePower reporting which would otherwise be misleading.
+                // Said differently, we report that we didn't distribute anything, and this is indeed consistent with the network state.
+                contextData.addDistributedActivePower(-totalDistributedActivePower);
+                return new OuterLoopResult(this, OuterLoopStatus.FAILED, statusText);
+            }
+            default -> throw new IllegalArgumentException("Unknown slackDistributionFailureBehavior");
+        }
+    }
+
+    private static String mismatchesToString(Map<LfArea, Double> remainingMismatchMap) {
+        return remainingMismatchMap.entrySet().stream()
+                .sorted(Comparator.comparing(entry -> entry.getKey().getId()))
+                .map(entry -> String.format(Locale.US, "%s: %.2f MW", entry.getKey().getId(), entry.getValue() * PerUnit.SB))
+                .collect(Collectors.joining(", "));
+    }
+
+    private double getInterchangeMismatch(LfArea area) {
+        return area.getInterchangeTarget() - area.getInterchange();
+    }
+
+    private double getInterchangeMismatchWithSlack(LfArea area, double slackBusActivePowerMismatch) {
+        return area.getInterchange() - area.getInterchangeTarget() + getSlackInjection(area, slackBusActivePowerMismatch);
```

Comment:
It is confusing I agree, in dae81198cb9c57278031756521f81e895fe80804 I used the same convention and added a minus sign to the sum of mismatch without slack when it is used.
(the sign is opposite because getInterchangeMismatchWithSlack is used to distribute the mismatch of areas inside them and getInterchangeMismatch to distribute it outside)